src/mscorlib/src/System/Runtime/CompilerServices/TaskAwaiter.cs

```diff
@@ -113,14 +114,19 @@ public void GetResult()
         /// prior to completing the await.
         /// </summary>
         /// <param name="task">The awaited task.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static void ValidateEnd(Task task)
         {
             // Fast checks that can be inlined.
             if (task.IsWaitNotificationEnabledOrNotRanToCompletion)
             {
                 // If either the end await bit is set or we're not completed successfully,
                 // fall back to the slower path.
-                HandleNonSuccessAndDebuggerNotification(task);
+                var exception = HandleNonSuccessAndDebuggerNotification(task);
+                if (exception != null)
+                {
+                    throw exception;
+                }
```

Comment:
Why did you change this?

Reply:
To throw in-place rather than deeper in the stack

Reply:
Will output the asm, check the before and after - hopefully its not terrible

Reply:
Main change looks to be
```diff
+       mov      rcx, rax
+       call     CORINFO_HELP_THROW
...
        call     TaskAwaiter:HandleNonSuccessAndDebuggerNotification(ref)
-       nop      		
+       test     rax, rax
+       jne      SHORT G_M53132_IG04
```